pkg/login/social/grafana_com_oauth.go

```diff
@@ -60,12 +63,17 @@ func (s *SocialGrafanaCom) UserInfo(client *http.Client, _ *oauth2.Token) (*Basi
 		return nil, fmt.Errorf("Error getting user info: %s", err)
 	}
 
+	// on login we do not want to display the role from the external provider
```

Comment:
I would move this comment one line below (just above the conditional checking `skipOrgRoleSync` value). I'd also reword it to something like 
```suggestion
	// do not sync the role from the external provider
```
to make it clearer that it isn't just about what we display but also about how user role is determined on the backend.